gcp/data-service/src/DataWriter.ts

```diff
@@ -1,6 +1,44 @@
-import { IDataWriter, DataRow, DataContext } from './types';
+import {
+  IDataWriter,
+  DataRow,
+  DataContext,
+  GenericDataConfig,
+  TokenDataConfig,
+} from './types';
 import { Knex } from 'knex';
 
+export class TokenBalanceDataWriter implements IDataWriter {
+  public async write(
+    db: Knex,
+    context: DataContext,
+    rows: DataRow[]
+  ): Promise<void> {
+    const query = db
+      .insert(
+        rows.map((v) => {
+          const dataConfig = v.dataConfig as TokenDataConfig;
+
+          return {
+            token_address: v.contractAddress,
+            network: v.networkId,
+            timestamp: context.timestamp,
+            block_number: v.blockNumber,
+            decimals: dataConfig.decimals,
+            value: v.value.toString(),
+          };
+        })
+      )
+      .into(context.tableName)
+      .onConflict(['strategy_id', 'variable', 'network', 'timestamp']);
```

Comment:
Are these the right column names for on conflict?

Reply:
fixed